File: src/models/simple-transaction.ts
import Entity from "./entity"

export default class SimpleTransaction extends Entity<SimpleTransaction> {

    constructor(
        public date: string, 
        public description: string, 
        public accountId: number,
        public amount: number,
        public isActive: boolean, 
        public isRecurrent: boolean,
        public id: number | null) {
        super(id, isActive);
    }
}

File: Fin.Api/Models/SimpleTransaction.cs
namespace Fin.Api.Models;

public class SimpleTransaction
{
    public int? Id { get; set; }
    public DateOnly Date { get; set; }
    public required string Description { get; set; }
    public int AccountId { get; set; }
    public decimal Amount { get; set; }
    public bool IsRecurrent { get; set; }
    public bool IsActive { get; set; }
}
